### DISCUSSION TITLE
Mysql database - step-ca

### DISCUSSION THREAD

**Discussion by fa-droussat** [smallstep/certificates#1213] [Q&A]
Hello ! 

I configured step-ca to use Mysql database. So I noticed that each certificate renewal added new rows in the following tables:
- acme_authzs
- acme_certs
- acme_challenges
- acme_orders
- acme_serial_certs_index
- nonces
- x509certs
- x509cert_data

Can I delete obsolete data without affecting my CA?

Thanks for your help!!

**Comment by hslatman** [comment]
Hey @fa-droussat, for most use cases it's safe to delete data from those tables, indeed. If you intend to use revocation, it's advisable to not delete all data in the tables, but only the data that's considered outdated (e.g. older than the certificate lifetime). At the moment properties that can be used to selectively delete data are not available, as all of our data storage implementations use a K/V model. It's possible to write your own Go program to delete tables, an example of which is given here: https://github.com/smallstep/certificates/issues/473#issuecomment-780328984. The corresponding [issue](https://github.com/smallstep/certificates/issues/473) also has some additional guidance.
